src/Tizen.NUI.Components/Controls/NumberPicker.cs

```diff
@@ -0,0 +1,504 @@
+
+                float currentScrollPosition = -(ScrollingDirection == Direction.Horizontal ? ContentContainer.CurrentPosition.X : ContentContainer.CurrentPosition.Y);
+                panAnimationDelta = (velocityOfLastPan * decelerationRate) / (1 - decelerationRate);
+                panAnimationDelta = velocity > 0 ? -panAnimationDelta : panAnimationDelta;
+
+                float destination = -(panAnimationDelta + currentScrollPosition);
+                //Animation destination has to center of the item.
+                float align = destination % itemHeight;
+                destination -= align;
+                destination -= startScrollOffset;
+
+                float adjustDestination = AdjustTargetPositionOfScrollAnimation(destination);
+
+                float maxPosition = ScrollAvailableArea != null ? ScrollAvailableArea.Y : ScrollAvailableArea.Y;
+                float minPosition = ScrollAvailableArea != null ? ScrollAvailableArea.X : 0;
+
+                if (destination < -maxPosition || destination > minPosition)
+                {
+                    panAnimationDelta = velocity > 0 ? (currentScrollPosition - minPosition) : (maxPosition - currentScrollPosition);
+                    destination = velocity > 0 ? minPosition : -maxPosition;
+
+                    if (panAnimationDelta == 0)
+                    {
+                        panAnimationDuration = 0.0f;
+                    }
+                    else
+                    {
+                        panAnimationDuration = (float)Math.Log((panAnimationDelta * logValueOfDeceleration / velocityOfLastPan + 1), decelerationRate);
+                    }
+                }
+                else
+                {
+                    panAnimationDuration = (float)Math.Log(-DecelerationThreshold * logValueOfDeceleration / velocityOfLastPan) / logValueOfDeceleration;
+
+                    if (adjustDestination != destination)
+                    {
+                        destination = adjustDestination;
+                        panAnimationDelta = destination + currentScrollPosition;
+                        velocityOfLastPan = Math.Abs(panAnimationDelta * logValueOfDeceleration / ((float)Math.Pow(decelerationRate, panAnimationDuration) - 1));
+                        panAnimationDuration = (float)Math.Log(-DecelerationThreshold * logValueOfDeceleration / velocityOfLastPan) / logValueOfDeceleration;
+                    }
+                }
+
+                customScrollAlphaFunction = new UserAlphaFunctionDelegate(CustomScrollAlphaFunction);
+                animation.DefaultAlphaFunction = new AlphaFunction(customScrollAlphaFunction);
+                animation.Duration = (int)panAnimationDuration;
+                animation.AnimateTo(ContentContainer, (ScrollingDirection == Direction.Horizontal) ? "PositionX" : "PositionY", (int)destination);
+                animation.Play();
+            }
+        }
+    }
+}
```

Comment:
could you add new line at the end of file?
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
Add new line. done.